build/build-pr-ci.yml

```diff
@@ -64,24 +41,50 @@ steps:
       dotnet build $(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI.AzureAISearch/WebJobs.Extensions.OpenAI.AzureAISearch.csproj --configuration $(config) -p:Version=$(fakeWebJobsPackageVersion) -p:AzureAISearchVersion=$(fakeWebJobsPackageVersion)
       dotnet build $(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI.CosmosDBSearch/WebJobs.Extensions.OpenAI.CosmosDBSearch.csproj --configuration $(config) -p:Version=$(fakeWebJobsPackageVersion) -p:CosmosDBSearchVersion=$(fakeWebJobsPackageVersion)
+      
       sudo cp "$(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.$(fakeWebJobsPackageVersion).nupkg" ~/NuGetPackagesLocal
       sudo cp "$(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI.AzureAISearch/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.AzureAISearch.$(fakeWebJobsPackageVersion).nupkg" ~/NuGetPackagesLocal
       sudo cp "$(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI.Kusto/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.Kusto.$(fakeWebJobsPackageVersion).nupkg" ~/NuGetPackagesLocal
       sudo cp "$(System.DefaultWorkingDirectory)/src/WebJobs.Extensions.OpenAI.CosmosDBSearch/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.CosmosDBSearch.$(fakeWebJobsPackageVersion).nupkg" ~/NuGetPackagesLocal
 
-      sleep 10s
-
       dotnet nuget add source ~/NuGetPackagesLocal --configfile  ~/.nuget/NuGet/NuGet.Config
       dotnet nuget list source --configfile  ~/.nuget/NuGet/NuGet.Config
 
-      sleep 10s
-
       dotnet build --configuration $(config) /p:WebJobsVersion=$(fakeWebJobsPackageVersion) /p:Version=$(fakeWebJobsPackageVersion) /p:AzureAISearchVersion=$(fakeWebJobsPackageVersion) /p:KustoVersion=$(fakeWebJobsPackageVersion) --configfile ~/.nuget/NuGet/NuGet.Config
     displayName: Create/Copy NuGet Package to local directory and Build
 
+  - task: DotNetCoreCLI@2
+    displayName: 'Publish Project'
+    inputs:
+      command: publish
+      publishWebProjects: false
+      projects: '$(System.DefaultWorkingDirectory)/samples/chat/csharp-ooproc/ChatBot.csproj'
+      arguments: '--configuration $(config) --output $(build.artifactstagingdirectory)'
+
+  - task: AzureCLI@2
+    inputs:
+      azureSubscription: $(serviceConnectionName)
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: 'az functionapp start --name $(functionAppName) --resource-group $(resourceGroupName)'
+
+  - task: AzureFunctionApp@2
+    inputs:
+      connectedServiceNameARM: $(serviceConnectionName)
+      appType: 'functionApp'
+      appName: $(functionAppName)
+      package: '$(build.artifactstagingdirectory)/csharp-ooproc.zip'
+      deploymentMethod: 'auto'
+
   - script: dotnet test --configuration $(config) --no-build --verbosity normal
+    env:  
+      FUNC_KEY: $(FUNC_KEY) 
```

Comment:
Where is this defined? Can we add a comment explaining what this is for and where it is configured?

Reply:
this is defined in pipeline variables and value is kept as secret, added a comment for the same.